tests/models/test_workspace_role.py

```diff
@@ -14,6 +15,50 @@
 )
 
 
+def test_has_no_history(session):
+    owner = UserFactory.create()
+    user = UserFactory.create()
+
+    workspace = Workspaces.create(RequestFactory.create(creator=owner))
+    workspace_role = WorkspaceRoles.add(user, workspace.id, "developer")
+    audit_events = (
+        session.query(AuditEvent)
+        .filter(AuditEvent.resource_id == workspace_role.id)
```

Comment:
You should be able to filter by action here as well:

```suggestion
        .filter(AuditEvent.resource_id == workspace_role.id, AuditEvent.action == "create")
```

Then, you won't need to do the list comprehension afterwards.